### PR TITLE
[SPARK-49582][PYTHON][CONNECT] Improve "dispatch_window_method" utility and docstring

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
- Remove the unreachable exception from "dispatch_window_method".
- Improve docstrings.

### Why are the changes needed?
Cleaner code and better documentation.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existing tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: python/pyspark/sql/utils.py [hunk 1]
```diff
@@ -399,11 +399,13 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
         if is_remote() and "PYSPARK_NO_NAMESPACE_SHARE" not in os.environ:
             from pyspark.sql.connect.window import Window as ConnectWindow
 
-            return getattr(ConnectWindow, f.__name__)(*args, **kwargs)
+            if isinstance(args[0], ConnectWindow):
```
Comment (zhengruifeng) [inline python/pyspark/sql/utils.py:402]: not sure whether need it, since all `Window` methods are static

Comment (zhengruifeng) [reply python/pyspark/sql/utils.py:402]: LGTM if it works

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/utils.py (hunk 1 above)
Comment (xinrong-meng) [reply python/pyspark/sql/utils.py]: Thank you for pointing it out! You are right. I removed the unreachable exception and adjusted the docstring.

**Review by HyukjinKwon — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by xinrong-meng** [comment]
Thank you all!

**Pull request closed without merging**